tests/regression/src/lib.rs

```diff
@@ -35,6 +35,20 @@ pub mod git {
             .map(|s| s.to_string())
             .unwrap_or_default() // This will return an empty string if the Option is None
     }
+
+    pub fn is_mainline(commit1: &str) -> bool {
```

Comment:
It's probably worth including a few more comments.

If `!output.status.success()`, does that mean the commit is detached?

And maybe include a note about why we have to handle `* main` in addition to `main`.